### PR TITLE
Remove VERSION file in che-docs; keep antora-playbook.yml up to date instead #18864

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

Instead of manual updates like https://github.com/eclipse/che-docs/commit/799956599b07de5634f819579381314b051e607f

This change will allow playbook to be updated with every Che release that invokes make-release.sh (manually or via GH action).

* d977f1f2 - 18864 remove VERSION file; update antora-playbook.yml instead nboldt@redhat.com (35 seconds ago)
* 630c41b1 - reorder stuff; switch from NOCOMMIT=0 to docommit=1 nboldt@redhat.com (35 seconds ago)
* 049c4ab7 - (HEAD -> gh18864, origin/gh18864) clean up usage instructions nboldt@redhat.com (35 seconds ago)

Change-Id: I1b6c60c4b1949018bb7ce5b38bde51b16ab9d819
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18864

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by themr0c — APPROVED** [review]

**Comment by nickboldt** [comment]
Occurs to me that the logic here will set:
* prod-prev-ver to 7.25 when prod-ver is 7.25.1
* prod-prev-ver to 7.25 when prod-ver is 7.25.2
* prod-prev-ver to 7.25 when prod-ver is 7.26.0
* prod-prev-ver to 7.26 when prod-ver is 7.26.1
* etc.

If that's not good, I can instead add another line that makes sure that {{prod-prev-ver}} is always one less - {{prod-ver--}}.

Which is better?

**Comment by themr0c** [comment]
```
 #    prod-ver-major: 7 [never changes] 
```
Hopefully one day this statement will become erroneous.

**Comment by themr0c** [comment]
We'll need to relaunch the build after publication of :
* https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-kubernetes/deploy/olm-catalog/eclipse-che-preview-kubernetes/7.25.2/eclipse-che-preview-kubernetes.v7.25.2.clusterserviceversion.yaml
* https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/7.25.2/eclipse-che-preview-openshift.v7.25.2.clusterserviceversion.yaml

**Comment by nickboldt** [comment]
> Hopefully one day this statement will become erroneous.

Don't hope for that! No one wants a Che 8 / CRW 3. cc: @l0rd

**Comment by themr0c** [comment]
Usage of `prod-prev-ver` is always upgrade path. I need the advice of an expert there:

```
modules/installation-guide/partials/proc_upgrading-che-using-the-cli-management-tool-in-restricted-environment.adoc:14:* A running instance version {prod-prev-ver} of {prod}, installed using the CLI management tool on the same instance of {platforms-name}, with the {prod-cli} `--installer operator` method, in the `_<{prod-namespace}>_` project. See xref:installing-che-in-a-restricted-environment.adoc[].
modules/installation-guide/partials/proc_upgrading-che-using-the-cli-management-tool-in-restricted-environment.adoc:22:. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the Git repositories.
modules/installation-guide/partials/proc_upgrading-che-using-the-cli-management-tool-in-restricted-environment.adoc:24:. Stop all workspaces in the {prod-short} {prod-prev-ver} instance.
modules/installation-guide/partials/proc_upgrading-che-using-the-cli-management-tool.adoc:20:. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the Git repositories.
modules/installation-guide/partials/proc_upgrading-che-using-the-cli-management-tool.adoc:22:. Shut down all workspaces in the {prod-short} {prod-prev-ver} instance.
modules/installation-guide/partials/assembly_upgrading-che-using-the-cli-management-tool-in-restricted-environment.adoc:8:This section describes how to upgrade {prod} using the CLI management tool in restricted environment. The upgrade path supports minor version update, from {prod-short} version {prod-prev-ver} to version {prod-ver}.
modules/installation-guide/partials/assembly_upgrading-che-using-the-cli-management-tool-in-restricted-environment.adoc:14:* A running instance version {prod-prev-ver} of {prod}, installed using the CLI management tool on the same instance of {platforms-name}, with the {prod-cli} `--installer operator` method, in the `_<{prod-namespace}>_` project. See xref:installing-che-in-a-restricted-environment.adoc[].
modules/installation-guide/partials/proc_uninstalling-che-after-operatorhub-installation-using-openshift-cli.adoc:58:{prod-id}.v{prod-ver}   {prod}   {prod-ver}     {prod-id}.v{prod-prev-ver}   Succeeded
modules/installation-guide/partials/assembly_upgrading-che.adoc:8:This chapter describes how to upgrade a {prod-short} instance from version {prod-prev-ver} to {prod-short} {prod-ver}.
modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc:108:. Add a {prod-short} {prod-prev-ver} recipe content to the {prod-short} {prod-ver} devfile as `referenceContent`:
modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc:112:.. Set the type from the original {prod-short} {prod-prev-ver} configuration. The following is an example of the resulting file:
```

**Comment by nickboldt** [comment]
This worked for creating 7.26.x branch and 7.26.0 release; seems to be having merge conflicts working in the 7.25.x branch as that branch has deviated from master branch. 

So perhaps we need to delete and recreate the branch for each 7.yy.z release? 

AFAICT the only value in having the .x branches is so we have a place from which to create tags, so that tagging can happen w/o having to wait for a PR merge. 

Questions:

* [ ] is prod-prev-ver always equal to prod-ver -1 (7.yy-1) ? Or should it be previous z release (7.yy.zzz-1) ? eg., for prod-ver = **7.26.1**, is prod-prev-ver **7.25** or **7.26.0**? 
*  [x] do we care about the .x branch content? if not, destroying them and recreating them simply so we can make the tags seems reasonable to me
* [x] should we move the che-doc update to the playbook file in the che-release process such that it's tied to the completion of the che-operator release? Seems like yes, as we need https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-kubernetes/deploy/olm-catalog/eclipse-che-preview-kubernetes/7.25.1/eclipse-che-preview-kubernetes.v7.25.1.clusterserviceversion.yaml to exist before we can reference it from the docs. Thus we need a new action which will trigger when changes happen in master branch: https://github.com/eclipse/che-operator/pull/657

**Comment by mkuznyetsov** [comment]
consider the case, if we have to make a bugfix release for "previous" version of Che, say 7.25.3, while we already have 7.26.0 up and referenced in master branch. This script will then change the values back to 7.25.x/7.24?

**Comment by themr0c** [comment]
Delete and recreate a released branch is not an option, as we are maintaining released branches: they contain a selection of backports for downstream, and some parts diverge: version numbers, content of generated docs, some specific fixes. 

To update a maintenance branch: start a feature branch from the maintenance branch, PR to the same maintenance branch. Master branch should not be in the loop.

**Comment by nickboldt** [comment]
So instead of `FORCE_NEW_BRANCH` you'd prefer a flow where if branch 7.yy.x exists, we create a merge request PR from 7.yy.x -> 7.yy.x? The only thing that would be changed in there is the value of prod-ver bumping from 7.yy.z to 7.yy.z+1, which... since the live docs are published only from master... would be a meaningless change anyway. Correct? 

OR, easier, we say "if branch exists, it was created for the 7.yy.0 release, so there's no need to push more commits from master (or 7.yy.x) into that .x branch unless a doc dev is explicitly cherry-picking. 

Therefore I could just remove the `FORCE_NEW_BRANCH` option entirely, and if branch already exists, we quietly skip over the branch creation step when doing a 7.yy.1 or 7.yy.2 bugfix release. 

WDYT?

**Comment by themr0c** [comment]
> "if branch exists, it was created for the 7.yy.0 release, so there's no need to push more commits from master (or 7.yy.x) into that .x branch unless a doc dev is explicitly cherry-picking."

I agree.

> The only thing that would be changed in there is the value of prod-ver bumping from 7.yy.z to 7.yy.z+1, which... since the live docs are published only from master... would be a meaningless change anyway. Correct?

Meaningless as "not published in the official docs website. It won't appear anywhere else than in this file on GitHub"

**Review by benoitf — APPROVED** [review]

File: make-release.sh [hunk 1]
```diff
@@ -74,44 +120,43 @@ fi
...
+  git tag "${VERSION}"
+  git push origin "${VERSION}" || true
 fi
-# bump the z digit
-[[ $VERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.7.1, get BASE=7.7, NEXT=2
-NEXTVERSION_Z="${BASE}.${NEXT}-SNAPSHOT"
-bump_version ${NEXTVERSION_Z} ${BRANCH}
+
+# cleanup 
+rm -fr /tmp/$tmpdir
```
Comment (benoitf) [inline make-release.sh]: minor: missing EOF

**Review by rkratky — APPROVED** [review]

**Review by MichalMaler — APPROVED** [review]

File: make-release.sh (hunk 1 above)
Comment (themr0c) [inline make-release.sh]:
```suggestion
rm -fr /tmp/$tmpdir

```

**Pull request merged**